### PR TITLE
Add test for useEffect ordering with useMemo

### PR DESCRIPTION
**Pull request by andrewiggins**
Adds a test for #3354

### CODE REVIEW

**Review by JoviDeCroock — APPROVED** [review]

File: hooks/test/browser/combinations.test.js [hunk 1]
```diff
@@ -298,4 +299,47 @@ describe('combinations', () => {
...
+	it('should run effects child-first even for children separated by memoization', () => {
+		let ops = [];
+
+		/** @type {() => void} */
+		let updateChild;
+		/** @type {() => void} */
+		let updateParent;
+
+		function Child() {
+			const [_, setCount] = useState(0);
```
Comment (JoviDeCroock) [inline hooks/test/browser/combinations.test.js:312]:
```suggestion
			const [, setCount] = useState(0);
```

File: hooks/test/browser/combinations.test.js [hunk 2]
```diff
@@ -298,4 +299,47 @@ describe('combinations', () => {
...
+			updateChild = () => setCount(c => c + 1);
+			useEffect(() => {
+				ops.push('child effect');
+			});
+			return <div>Child</div>;
+		}
+
+		function Parent() {
+			const [_, setCount] = useState(0);
```
Comment (JoviDeCroock) [inline hooks/test/browser/combinations.test.js:321]:
```suggestion
			const [, setCount] = useState(0);
```

**Review by JoviDeCroock — APPROVED** [review]
Should resolve the issues

File: hooks/test/browser/combinations.test.js (hunk 1 above)
Comment (andrewiggins) [reply hooks/test/browser/combinations.test.js:312]: Thanks! TIL this was possible lol

**Pull request merged**